tests/infrastructure/golden_images/test_common_templates_data_volumes.py

```diff
@@ -102,56 +128,42 @@ def test_vm_with_existing_dv(data_volume_scope_function, vm_from_template_with_e
 
 
 @pytest.mark.parametrize(
-    "golden_image_data_volume_scope_function, vm_from_golden_image",
+    "fedora_vm_from_data_source",
     [
         pytest.param(
             {
-                "dv_name": FEDORA_LATEST_OS,
-                "image": FEDORA_LATEST.get("image_path"),
-                "storage_class": HOSTPATH_CSI_BASIC,
-                "dv_size": FEDORA_LATEST.get("dv_size"),
-            },
-            {
-                "ocs_storage_class": False,
+                "set_storage_class": True,
             },
             marks=pytest.mark.polarion("CNV-5529"),
         ),
     ],
     indirect=True,
 )
 def test_vm_dv_with_different_sc(
-    fail_test_if_no_ocs_sc,
-    fail_if_no_hostpath_csi_basic_sc,
-    vm_from_golden_image,
+    fedora_vm_from_data_source,
 ):
     # VM cloned PVC storage class is different from the original golden image storage class
-    running_vm(vm=vm_from_golden_image)
+    running_vm(vm=fedora_vm_from_data_source)
 
 
 @pytest.mark.parametrize(
-    "golden_image_data_volume_scope_function, vm_from_golden_image",
+    "fedora_vm_from_data_source",
     [
         pytest.param(
             {
-                "dv_name": FEDORA_LATEST_OS,
-                "image": FEDORA_LATEST.get("image_path"),
-                "dv_size": FEDORA_LATEST.get("dv_size"),
-                "storage_class": py_config["default_storage_class"],
-            },
-            {
-                "ocs_storage_class": True,
+                "set_storage_class": False,
             },
             marks=pytest.mark.polarion("CNV-7752"),
         ),
     ],
     indirect=True,
 )
-def test_vm_from_golden_image_missing_default_storage_class(
+def test_vm_from_data_source_missing_default_storage_class(
     removed_default_storage_classes,
-    vm_from_golden_image,
+    fedora_vm_from_data_source,
 ):
-    vm_from_golden_image.start()
-    volume_status = vm_from_golden_image.instance.status.volumeSnapshotStatuses[0]
+    fedora_vm_from_data_source.start()
```

Comment:
if we wait for VM to start it actually gets started and running,  so this test will fail in that case. 
a slight pause or wait should not make it fail. 

E         - PVC not found
E         + No VolumeSnapshotClass: Volume snapshots are not configured for this StorageClass [hostpath-csi-basic] [dv-disk]

Reply:
Updating the test feels out of scope for this PR, since the main focus here is on storage class changes. It might be clearer to address test adjustments in a separate PR, so the context of this PR remains focused.